docs/DeveloperGuide.adoc

```diff
@@ -147,14 +149,206 @@ The `Storage` component,
+
+image::ClashSequenceDiagram.png[]
+
+===== Aspect: How to deal with different user inputs for different checking
+
+* **Alternative 1 (current choice):** Create three different sub-classes extending `ClashCommand` class. `ClashCommand` will deal with differentiating cases for checking and create Command object accordingly.
+** Pros: Easy to implement.
+** Cons: some of the code are different to be abstracted. (repeted codes in part of the two sub-classes to keep low coupling) May need to find a way to improve it in the future.
+* **Alternative 2:** Only 1 `ClashCommand` class. The cases of checking are determined inside `ClashCommand`.
+** Pros:
+** Cons: Violates Single Responsibility Principle. `ClashCommand` is doing part of the parsing work.
+
+===== Aspect: String representation of clashig time slots
+
+* **Alternative 1 (current choice):** Use a list to store time slot index and use them to create a temporary `Module` object.
+** Pros: Can reuse `getTimeSlotToString()` to obtain the string representation
+** Cons: Redundant objects are created. More memory space used.
+// end::clash[]
+
+//tag::find[]
+=== Find feature
+==== Implementation
+
+The find feature is facilitated by `FindCommand` class. The `FindCommand` constructor takes a `List` of `Predicates`, which can be `NameContainsKeywordsPredicate`, `ModuleContainsKeywordsPredicate`, or `AppealContainsKeywordsPredicate`
+
+image::PredicateClassDiagram.png[]
+
+Find commands only retrieve information from model but do not alter anything in the existing lists.
+
+Below shows how the Find Command mechanism behaves at each step.
+
+Step 1. The user inputs `find prefix/[KEYWORDS]...`.
+
+Step 2. The `FindCommandParser` will be called first to check which of the three lists to search by checking the prefixes.
+`NameContainsKeywordsPredicate`, `ModuleContainsKeywordsPredicate`, and `AppealContainsKeywordsPredicate` will be created accordingly which stores all the keywords for the same list.
+
+Step 3. A new `FindCommand` is created.
+
+[NOTE]
+Even there is no keyword for a particular `Predicate`, a new `Predicate` object with an empty `List` of keywords will be passed in to create a new `FindCommand`.
+Eg. input: `find s/bob`. A new `FindCommand` is created with a `NameContainsKeywordsPredicate` with a `List` of keywords containing `bob`, and `ModuleContainsKeywordsPredicate`, and `AppealContainsKeywordsPredicate` with empty `List` of keywords.
+
+Step 4. `FindCommand.execute()` is called. Each `Predicate` object is examined and calls `Predicate.test()`. `Model#updateStudentList()`, `Model#updateModuleList()`, or `Module#updateAppealList()` is called accordingly.
+
+Step 5. A `CommandResult` object is returned, which contains message about the number of items found in the target list. The target list only shows the item contains keyword.
+
+The following sequence diagram shows how the undo operation works:
+
+image::FindSequenceDiagram.png[]
```

Comment:
You can use nckp: NameContainsKeywordsPredicate and then use nckp at updateFilteredStudentList(nckp) call